### PR TITLE
fix: make binaries work from npx

### PR DESCRIPTION
**Pull request by izaera**
Right now, `npx @liferay/npm-bundler` fails because it cannot find the executable `.js` script.

We have made a very deep, intelligent, exhaustive, and thorough investigation (basically just peeking [how `@angular/cli` declares its binary](https://github.com/angular/angular-cli/blob/4169e303c3075df3bdb8c7f500a2d034d6532537/packages/angular/cli/package.json#L7)) and have come to the conclusion that adding `./` to the left and maybe removing the `.js` may help.

Unfortunately we need to merge this PR and release a new version to test this :-(.

### CODE REVIEW

**Comment by izaera** [comment]
I'm going to merge this since we don't have any other option to make sure it is correct.

**Comment by izaera** [comment]
According to [npx's documentation](https://www.npmjs.com/package/npx) the problem may be more in the name of the executable than on the path itself. Going to try to investigate it locally...

**Comment by wincent** [comment]
> According to [npx's documentation](https://www.npmjs.com/package/npx) the problem may be more in the name of the executable than on the path itself. Going to try to investigate it locally...

It works fine with `npx @liferay/npm-scripts` though, that has exactly the same naming/structure, except `./bin` instead of `bin`.

**Comment by wincent** [comment]
And it [uses the `.js` extension](https://github.com/liferay/liferay-frontend-projects/blob/993b70b9fc23165b023b531c6c00118ba9189d2f/projects/npm-tools/packages/npm-scripts/package.json#L4), so I am not sure it will work without it (it might, but I have no past experience with omitting the extension).

**Comment by izaera** [comment]
I think the extension makes no difference in the end, because Windows' node creates a `.bat`|`.cmd` file under the hood to be able to execute things so I think it takes care of the extension for us...

However, I'm not sure now if this is the real error (see [this comment](https://github.com/liferay/liferay-frontend-projects/issues/217#issuecomment-718724923)).

**Comment by wincent** [comment]
Can you leave the extension the way it was then? That, at least, is useful for users of IDEs who can hit a binding when their cursor is on the filename in order to jump to that file.

Even if this doesn't fix that other error, seems like the change from `bin` to `./bin` should still go in.

**Comment by wincent** [comment]
Sounds good.

**Comment by izaera** [comment]
Seeing that ["my" error](https://github.com/liferay/liferay-frontend-projects/issues/217#issuecomment-718724923) only happens when `npx` is used without any `node_modules` at all and, as that situation doesn't make sense for the tool and seems complex to fix, I won't be fixing that.

Instead, @p2kmgcl will [workaround it in Fragments CLI](https://github.com/liferay/generator-liferay-fragments/pull/176). In case that doesn't work, we will change the bundler to let users directly invoke its [main method](https://github.com/liferay/liferay-frontend-projects/blob/993b70b9fc23165b023b531c6c00118ba9189d2f/projects/js-toolkit/packages/npm-bundler/src/index.ts#L19) programmatically. That should be easier to hack and more useful for everybody.

So I guess we can merge this if you agree @wincent .

**Comment by wincent** [comment]
> So I guess we can merge this if you agree @wincent .

I do, because:

> Even if this doesn't fix that other error, seems like the change from `bin` to `./bin` should still go in.

**Review by wincent — APPROVED** [review]
Don't know if the title should be edited before merging. Seeing as it doesn't seem to actually fix the problem. Maybe it should be "refactor(js-toolkit): make `bin` properties consistent".

**Comment by izaera** [comment]
OK, we shouldn't merge this then, because the commit messages are also misleading. I will resend it.

**Pull request closed without merging**
